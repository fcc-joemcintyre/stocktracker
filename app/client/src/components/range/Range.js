import PropTypes from 'prop-types';

const Item = ({ months, selected, text, onRangeChanged }) => (
  <div
    style={{
      display: 'inline-block',
      padding: '4px 12px',
      color: '#ffffff',
      backgroundColor: selected ? 'teal' : 'blue',
      borderLeft: months > 1 ? '1px solid #ffffff' : null,
    }}
    onClick={() => onRangeChanged (months)}
  >
    {text}
  </div>
);

Item.propTypes = {
  months: PropTypes.number.isRequired,
  selected: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  onRangeChanged: PropTypes.func.isRequired,
};

/*
 * Set of ranges displayed, allowing user to select one
 */
export const Range = ({ months, onRangeChanged }) => (
  <div style={{ display: 'flex' }}>
    <div style={{ border: '1px solid #7f7f7f', borderRadius: '8px', overflow: 'hidden' }}>
      <Item months={1} selected={months === 1} text='1m' onRangeChanged={onRangeChanged} />
      <Item months={3} selected={months === 3} text='3m' onRangeChanged={onRangeChanged} />
      <Item months={6} selected={months === 6} text='6m' onRangeChanged={onRangeChanged} />
      <Item months={12} selected={months === 12} text='1y' onRangeChanged={onRangeChanged} />
      <Item months={36} selected={months === 36} text='3y' onRangeChanged={onRangeChanged} />
    </div>
    <div flex='1 1 0' />
  </div>
);

Range.propTypes = {
  // months in range
  months: PropTypes.number,
  // function to call on range change
  onRangeChanged: PropTypes.func.isRequired,
};

Range.defaultProps = {
  months: 12,
};
