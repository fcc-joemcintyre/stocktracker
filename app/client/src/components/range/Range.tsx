type ItemProps = {
  months: number,
  selected: boolean,
  text: string,
  onRangeChanged: (months: number) => void,
};

const Item = ({
  months, selected, text, onRangeChanged,
}: ItemProps) => (
  <div
    style={{
      display: 'inline-block',
      padding: '4px 12px',
      color: '#ffffff',
      backgroundColor: selected ? 'teal' : 'blue',
      borderLeft: months > 1 ? '1px solid #ffffff' : undefined,
    }}
    onClick={() => onRangeChanged (months)}
  >
    {text}
  </div>
);

type RangeProps = {
  months?: number,
  onRangeChanged: (months: number) => void,
};

/*
 * Set of ranges displayed, allowing user to select one
 */
export const Range = ({
  months = 12, onRangeChanged,
}: RangeProps) => (
  <div style={{ display: 'flex' }}>
    <div style={{ border: '1px solid #7f7f7f', borderRadius: '8px', overflow: 'hidden' }}>
      <Item months={1} selected={months === 1} text='1m' onRangeChanged={onRangeChanged} />
      <Item months={3} selected={months === 3} text='3m' onRangeChanged={onRangeChanged} />
      <Item months={6} selected={months === 6} text='6m' onRangeChanged={onRangeChanged} />
      <Item months={12} selected={months === 12} text='1y' onRangeChanged={onRangeChanged} />
      <Item months={36} selected={months === 36} text='3y' onRangeChanged={onRangeChanged} />
    </div>
    <div style={{ flex: '1 1 0' }} />
  </div>
);
