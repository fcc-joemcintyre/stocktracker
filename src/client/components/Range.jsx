/**
 * Copyright (c) Joe McIntyre, 2016-2018
 * license: MIT (https://github.com/fcc-joemcintyre/stocktracker/LICENSE.txt)
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Outstanding requests and error status display.
 */
export class Range extends Component {
  constructor (props) {
    super (props);
    this.state = {
      months: props.months,
    };
    this.onRangeSelected = this.onRangeSelected.bind (this);
  }

  onRangeSelected (months) {
    if (months !== this.state.months) {
      this.setState ({ months });
      this.props.onRangeChanged (months);
    }
  }

  render () {
    const spans = [];
    for (const value of [1, 3, 6, 12, 36]) {
      spans.push (
        <span
          key={value}
          className={(value === this.state.months) ? 'selected' : ''}
          onClick={() => { this.onRangeSelected (value); }}
        >
          {(value < 12) ? `${value}m` : `${(value / 12)}y`}
        </span>
      );
    }

    return (
      <div className='rangeBox'>
        {spans}
      </div>
    );
  }
}

Range.propTypes = {
  // months in range
  months: PropTypes.number,
  // function to call on range change
  onRangeChanged: PropTypes.func.isRequired,
};

Range.defaultProps = {
  months: 12,
};
