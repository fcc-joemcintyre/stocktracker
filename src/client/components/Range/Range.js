import React from 'react';
import PropTypes from 'prop-types';

/*
 * Set of ranges displayed, allowing user to select one
 */
export const Range = ({ months, onRangeChanged }) => {
  const spans = [];
  for (const value of [1, 3, 6, 12, 36]) {
    spans.push (
      <span
        key={value}
        className={(value === months) ? 'selected' : ''}
        onClick={() => onRangeChanged (value)}
      >
        {(value < 12) ? `${value}m` : `${(value / 12)}y`}
      </span>
    );
  }

  return (
    <div className='rangeBox'>
      {spans}
    </div>
  );
};

Range.propTypes = {
  // months in range
  months: PropTypes.number,
  // function to call on range change
  onRangeChanged: PropTypes.func.isRequired,
};

Range.defaultProps = {
  months: 12,
};
