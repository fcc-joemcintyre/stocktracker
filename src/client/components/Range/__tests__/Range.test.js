/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Range } from '../Range';

test ('Range has default selection (1yr)', () => {
  const component = renderer.create (
    <Range onRangeChanged={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Range has set selection (3m)', () => {
  const component = renderer.create (
    <Range months={3} onRangeChanged={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Range has minimum selection (1m)', () => {
  const component = renderer.create (
    <Range months={1} onRangeChanged={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Range has minimum selection (36m)', () => {
  const component = renderer.create (
    <Range months={36} onRangeChanged={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
