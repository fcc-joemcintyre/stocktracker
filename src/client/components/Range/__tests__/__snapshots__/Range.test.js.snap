// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range has default selection (1yr) 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #7f7f7f",
        "borderRadius": "8px",
        "overflow": "hidden",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": null,
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      6m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "teal",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1y
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3y
    </div>
  </div>
  <div
    flex="1 1 0"
  />
</div>
`;

exports[`Range has minimum selection (1m) 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #7f7f7f",
        "borderRadius": "8px",
        "overflow": "hidden",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "teal",
          "borderLeft": null,
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      6m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1y
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3y
    </div>
  </div>
  <div
    flex="1 1 0"
  />
</div>
`;

exports[`Range has minimum selection (36m) 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #7f7f7f",
        "borderRadius": "8px",
        "overflow": "hidden",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": null,
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      6m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1y
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "teal",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3y
    </div>
  </div>
  <div
    flex="1 1 0"
  />
</div>
`;

exports[`Range has set selection (3m) 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "border": "1px solid #7f7f7f",
        "borderRadius": "8px",
        "overflow": "hidden",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": null,
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "teal",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      6m
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      1y
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "borderLeft": "1px solid #ffffff",
          "color": "#ffffff",
          "display": "inline-block",
          "padding": "4px 12px",
        }
      }
    >
      3y
    </div>
  </div>
  <div
    flex="1 1 0"
  />
</div>
`;
